PSModule/AlmLabMgmtClient/SDK/Handler/LabPollHandler.cs

```diff
@@ -0,0 +1,144 @@
+using PSModule.AlmLabMgmtClient.SDK.Interface;
+using PSModule.AlmLabMgmtClient.SDK.Request;
+using PSModule.AlmLabMgmtClient.SDK.Util;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PSModule.AlmLabMgmtClient.SDK.Handler
+{
+    public class LabPollHandler : PollHandler
+    {
+        private const string END_TIME = "end-time";
+        private const string START_TIME = "start-time";
+        private const string STATE = "state";
+        private const string COMPLETED_SUCCESSFULLY = "completed-successfully";
+        private const string RESERVATION_ID = "reservation-id";
+
+        private EventLogHandler _eventLogHandler;
+
+        public LabPollHandler(IClient client, string entityId) : base(client, entityId)
+        {
+        }
+
+        public LabPollHandler(IClient client, string entityId, int interval) : base(client, entityId, interval) { }
+
+        protected override async Task<bool> DoPoll()
+        {
+            bool ok = false;
+
+            Response res = await GetRunEntityData();
+            if (res.IsOK)
+            {
+                SetTimeslotId(res);
+                _eventLogHandler = new EventLogHandler(_client, _timeslotId);
+                if (!_timeslotId.IsNullOrWhiteSpace())
+                {
+                    ok = await base.DoPoll();
+                }
+            }
+            else
+            {
+                LogPollingError(res);
+            }
+            return ok;
+        }
+
+        protected override async Task<Response> GetResponse(bool logRequestUrl)
+        {
+            return await new PollAlmLabMgmtRunRequest(_client, _runId).Execute(logRequestUrl);
+        }
+
+        protected async override Task LogProgress(bool logRequestUrl)
+        {
+            await _eventLogHandler.Log(logRequestUrl);
+        }
+
+        protected override bool IsFinished(Response response)
+        {
+            bool ret = false;
+            try
+            {
+                string xml = response.ToString();
+                string endTime = Xml.GetAttributeValue(xml, END_TIME);
+                if (!endTime.IsNullOrWhiteSpace())
+                {
+                    string startTime = Xml.GetAttributeValue(xml, START_TIME);
+                    string currentRunState = Xml.GetAttributeValue(xml, STATE);
+                    _logger.LogInfo($"Timeslot {_timeslotId} is [{currentRunState}].\nRun start time: [{startTime}], Run end time: [{endTime}]");
+                    ret = true;

```

Comment:
Do we need to check if the `startTime` and `currentRunState` is not empty and valid before setting `ret` to `true`?

Reply:
The idea is very good, but currently we don't know if we will get always the same time format across all ALM server versions.  If the received value does not match the expected formats the code will enter an infinite loop.
If the received value is not a timestamp, the code will stop the polling and exit the loop, which is not the best behavior, but from the printed output the user can observe and report the issue.

Reply:
Got you. Thanks.